### PR TITLE
enum reflection

### PR DESCRIPTION
**Pull request by Tsche**

### CODE REVIEW

File: include/librepr/ctvi/ctvi.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+namespace librepr::ctvi {
+namespace detail {
+template <std::size_t... Idx>
+consteval auto sv_to_array(std::string_view str, std::index_sequence<Idx...>) noexcept
+    -> std::array<char, (sizeof...(Idx) + 1)> {
+  return {{str[Idx]..., '\0'}};
+}
+
+template <auto V>
+[[nodiscard]] constexpr auto get_ctvi() noexcept {
```
Comment (8ightfold) [inline include/librepr/ctvi/ctvi.h:15]: Gotta make all the ``__PRETTY_FUNCTION__`` related stuff to compiler independent

File: include/librepr/enum/accessor.h [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  using structure            = Structure;
+  using underlying           = std::underlying_type_t<T>;
+  constexpr static auto kind = Kind;
+
+private:
+  static consteval auto get_enum_names() {
+    if constexpr (structure::size == 0) {
+      return std::array<std::string_view, 0>{};
+    } else {
+      auto constexpr full_size = structure::invoke([]<typename... Range>() { return ((Range::size) + ...); });
```
Comment (8ightfold) [inline include/librepr/enum/accessor.h:26]: Should probably make a metafunction that does this and call ``get_enum_names`` with it

File: include/librepr/enum/accessor.h [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      auto buffer = std::array<std::string_view, full_size>{};
+
+      structure::for_each(
+          [offset = 0U]<typename Range>(auto& output) mutable {
```
Comment (8ightfold) [inline include/librepr/enum/accessor.h:30]: Avoid creating lambdas if possible, it's insanely slow

File: include/librepr/ctvi/ctvi.h [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  constexpr std::size_t start = signature.find(prefix) + prefix.size();
+  constexpr std::size_t end   = signature.find(suffix, start);
+
+  constexpr auto value = signature.substr(start, end - start);
+  return sv_to_array(value, std::make_index_sequence<value.size()>{});
+}
+}  // namespace detail
+
+template <auto V>
+constexpr inline auto value = detail::get_ctvi<V>();
```
Comment (8ightfold) [inline include/librepr/ctvi/ctvi.h:30]: Wrap this in a ``string_view``

File: include/librepr/enum/util.h [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  } else {
+    using underlying = std::underlying_type_t<T>;
+    // https://discourse.llvm.org/t/clang-16-notice-of-potentially-breaking-changes/65562
+    // GCC seems to warn for this as well
+    // TODO use this new diagnostic for search once https://github.com/llvm/llvm-project/issues/57176 is fixed
+
+    // library UB does not need to be diagnosed
+    // (https://eel.is/c++draft/expr.const#5.31) hence we can use std::bit_cast
+    // here to bypass the miscompile
+    // return __builtin_bit_cast(T, Value); //better debug output
```
Comment (8ightfold) [inline include/librepr/enum/util.h:39]: Why don't we use the builtin directly?

Comment (Tsche) [reply include/librepr/enum/util.h:39]: pretty sure that's gcc/clang specific. There's no obvious performance impact from not using it directly and if it fails here we're in big trouble anyway :P

**Pull request merged**